### PR TITLE
Adding JLBP-2 (Minimize API surface)

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**

### CODE REVIEW

File: library-best-practices/JLBP-2.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[JLBP-2] Minimize API surface
+-----------------------------
+
+- Try not to expose types from your dependencies.
+  - Your dependencies have their own rules about how often
+    they do major version bumps and make breaking changes,
+    and could be shorter than yours, which could cause a lot
```
Comment (elharo) [inline library-best-practices/JLBP-2.md]: ", and" --> . These might be shorter than yours which causes problems when you

File: library-best-practices/JLBP-2.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    of problems if you want to retain a stable surface for
+    longer than your dependencies. Historically this was a
+    big problem for Guava, which made a major version bump
+    every 6 months, although Guava no longer does this.
+- Use protected packages/classes when the API is internal
```
Comment (elharo) [inline library-best-practices/JLBP-2.md]: package protected classes?

File: library-best-practices/JLBP-2.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  only and should not be used by consumers.
+- Do not mark methods and classes public by default.
+  Assume non-public until a need is known.
+- Prefer fewer packages over more packages to avoid
+  unnecessarily publicizing internal and implementation details,
```
Comment (elharo) [inline library-best-practices/JLBP-2.md]: delete "and"

Comment (garrettjonesgoogle) [reply library-best-practices/JLBP-2.md]: I simplified it and removed "implementation" also.

File: library-best-practices/JLBP-2.md (hunk 2 above)
Comment (garrettjonesgoogle) [reply library-best-practices/JLBP-2.md]: Done.

File: library-best-practices/JLBP-2.md (hunk 1 above)
Comment (garrettjonesgoogle) [reply library-best-practices/JLBP-2.md]: I split the sentences as suggested, but used slightly different wording.

**Comment by garrettjonesgoogle** [comment]
Feedback addressed, PTAL

**Pull request merged**